### PR TITLE
Add local codecov check script

### PR DESCRIPTION
**Pull request by richardwxn**
Forked from codecov scripts from istio repo and did some minor changes. Currently there is no test available so the report generated would still be empty.

### CODE REVIEW

**Comment by ostromart** [comment]
Thanks for looking into this - I suggest doing this in two steps: copy the files without modification and add any changes in a followup PR. There's code in the repo so you should get some results now.

**Comment by richardwxn** [comment]
> Thanks for looking into this - I suggest doing this in two steps: copy the files without modification and add any changes in a followup PR. There's code in the repo so you should get some results now.

yes, now we are able to get some results already. Actually I already exclude some changes from this PR, so it only contains necessary change such as path, filenames etc.

**Comment by ostromart** [comment]
sg. would you mind splitting this up into two PR's? first would be just to copy scripts unmodified from istio. second would be to patch in any changes. this will create a record of any changes we've had to make.

**Comment by richardwxn** [comment]
> sg. would you mind splitting this up into two PR's? first would be just to copy scripts unmodified from istio. second would be to patch in any changes. this will create a record of any changes we've had to make.

create another PR to just copy scripts unmodified #32

File: go.mod [hunk 1]
```diff
@@ -13,6 +13,7 @@ require (
 	github.com/imdario/mergo v0.3.7 // indirect
 	github.com/jstemmer/go-junit-report v0.0.0-20190106144839-af01ea7f8024 // indirect
 	github.com/kr/pretty v0.1.0
+	github.com/jstemmer/go-junit-report v0.0.0-20190106144839-af01ea7f8024 // indirect
```
Comment (ostromart) [inline go.mod]: remove

File: go.sum [hunk 2]
```diff
@@ -263,6 +263,8 @@ github.com/wadey/gocovmerge v0.0.0-20160331181800-b5bfa59ec0ad h1:W0LEBv82YCGEtc
 github.com/wadey/gocovmerge v0.0.0-20160331181800-b5bfa59ec0ad/go.mod h1:Hy8o65+MXnS6EwGElrSRjUzQDLXreJlzYLlWiHtt8hM=
 github.com/ugorji/go/codec v0.0.0-20181204163529-d75b2dcb6bc8/go.mod h1:VFNgLljTbGfSG7qAOspJ7OScBnGdDN/yBr0sguwnwf0=
 github.com/xordataexchange/crypt v0.0.3-0.20170626215501-b2862e3d0a77/go.mod h1:aYKd//L2LvnjZzWKhF00oedf4jCCReLcmhLdhm1A27Q=
+github.com/wadey/gocovmerge v0.0.0-20160331181800-b5bfa59ec0ad h1:W0LEBv82YCGEtcmPA3uNZBI33/qF//HAAs3MawDjRa0=
+github.com/wadey/gocovmerge v0.0.0-20160331181800-b5bfa59ec0ad/go.mod h1:Hy8o65+MXnS6EwGElrSRjUzQDLXreJlzYLlWiHtt8hM=
```
Comment (ostromart) [inline go.sum]: duplicated?

File: scripts/codecov.sh [hunk 3]
```diff
@@ -35,22 +35,14 @@ GO_TOP=$(cd "$(dirname "$0")"/../../../..; pwd)
 
 export GOPATH=${GOPATH:-$GO_TOP}
 
-if [ "${1:-}" != "" ]; then
+if [[ "${1:-}" != "" ]]; then
```
Comment (ostromart) [inline scripts/codecov.sh]: should this be fixed also in istio/istio? 
that's what happens when you touch something ;-)

Comment (richardwxn) [reply scripts/codecov.sh]: thanks for reminder, I would fix related problems in istio/istio

File: go.mod (hunk 1 above)
Comment (richardwxn) [reply go.mod]: done

**Pull request merged**